internal/offline/zip.go

```diff
@@ -0,0 +1,254 @@
+package offline
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/models"
+	"github.com/google/osv-scanner/pkg/osv"
+)
+
+type ZipDB struct {
+	// the name of the database
+	Name string
+	// the url that the zip archive was downloaded from
+	ArchiveURL string
+	// whether this database should make any network requests
+	Offline bool
```

Comment:
Maybe we should name this `ShouldUpdate` or something similar ? That makes the intent a lot clearer, and currently as it's named it's a bit confusing in the context of this being inside an "offline" package.

Reply:
what do you think about renaming the package itself to something like `local`? I'm not too fussed about renaming the property either, I just felt that `Offline` is pretty clear on what should (or should not) happen and that it is conceivable that other features get added that involve talking to a network.

Reply:
`local` SGTM.